include/string_span.h

```diff
@@ -178,4 +213,15 @@ template <size_t Max = dynamic_range>
 using wzstring_builder = basic_zstring_builder<wchar_t, Max>;
 }
 
+// VS 2013 workarounds
+#ifdef _MSC_VER
+#if _MSC_VER <= 1800
+
+#pragma pop_macro("GSL_MSVC_HAS_TYPE_DEDUCTION_BUG") 
+#undef GSL_MSVC_HAS_TYPE_DEDUCTION_BUG 
+
```

Comment:
Good point. I'm going to clean up all the config macros in a separate pass shortly. Given no one else should be using it anyway it's harmless for now.